### PR TITLE
Enable testing arm64 Mac architecture in the CI.

### PR DESCRIPTION
**Pull request by h-mayorquin**
Moving forward with #3100.

### CODE REVIEW

**Comment by h-mayorquin** [comment]
Uh, that's great, it just... worked.

**Comment by h-mayorquin** [comment]
Ah, of course it was so easy, we did not have logic for running everything if the workflow files from github actions are changed. I added that.

**Comment by h-mayorquin** [comment]
Ok, this is ready. I also eliminated a warning from the CI to keep things tidy  🧹 🧹 🧹 (   x D )

**Comment by zm711** [comment]
I think the only potentially slow step would be if we do a bunch of torch things, since pytorch defaults to the Mac CPU instead of GPU, but I don't even know what the runner GPU story is.

File: src/spikeinterface/extractors/tests/test_neoextractors.py [hunk 1]
```diff
@@ -278,8 +278,8 @@ class Spike2RecordingTest(RecordingCommonTestSuite, unittest.TestCase):
 
 
 @pytest.mark.skipif(
-    version.parse(platform.python_version()) >= version.parse("3.10"),
-    reason="Sonpy only testing with Python < 3.10!",
+    version.parse(platform.python_version()) >= version.parse("3.10") or platform.system() == "Darwin",
+    reason="Sonpy only testing with Python < 3.10 and not supported on macOS!",
```
Comment (zm711) [inline src/spikeinterface/extractors/tests/test_neoextractors.py:282]: Is this an Arm issue?

Comment (h-mayorquin) [reply src/spikeinterface/extractors/tests/test_neoextractors.py:282]: Yes, want to give it a shot to have something that excludes intel cpus from mac? I think they stopped selling two years ago only so many people might still have them.

That said, I did not want to allocate extra effort to something that won't work in a year anyway (see the python restriction)

**Review by zm711 — APPROVED** [review]
For me this is good to go. Based on a google intel Macs officially went end of life in JUNE2024. So only testing on officially supported hardware seems fair to me.  And it looks like it successfully tested everything for complete tests so I think this should be good to go.

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**